Gems/Atom/RPI/Code/Tests/Model/ModelTests.cpp

```diff
@@ -1031,42 +1035,76 @@ namespace UnitTest
-                lodCreator.SetMeshIndexBuffer({
-                    indexBuffer,
-                    AZ::RHI::BufferViewDescriptor::CreateStructured(0, static_cast<uint32_t>(indicesCount), sizeof(uint32_t))
-                });
-            }
+            AZStd::copy(indices, indices + indexCount, reinterpret_cast<uint32_t*>(const_cast<uint8_t*>(indexBuffer->GetBuffer().data())));
+            lodCreator.SetMeshIndexBuffer({ indexBuffer,
+                                            AZ::RHI::BufferViewDescriptor::CreateStructured(
+                                                aznumeric_cast<uint32_t>(indexOffset), aznumeric_cast<uint32_t>(indexCount), sizeof(uint32_t)) });
+            AZStd::copy(
+                positions, positions + positionCount, reinterpret_cast<float*>(const_cast<uint8_t*>(positionBuffer->GetBuffer().data())));
+            lodCreator.AddMeshStreamBuffer(
+                AZ::RHI::ShaderSemantic(AZ::Name("POSITION")), AZ::Name(),
+                { positionBuffer,
+                  AZ::RHI::BufferViewDescriptor::CreateStructured(
+                      aznumeric_cast<uint32_t>(positionOffset), aznumeric_cast<uint32_t>(positionCount / 3), sizeof(float) * 3) });
 
-            {
-                AZ::Data::Asset<AZ::RPI::BufferAsset> positionBuffer = BuildTestBuffer(static_cast<uint32_t>(positionCount / 3), sizeof(float) * 3);
-                AZStd::copy(positions, positions + positionCount, reinterpret_cast<float*>(const_cast<uint8_t*>(positionBuffer->GetBuffer().data())));
-                lodCreator.AddMeshStreamBuffer(
-                    AZ::RHI::ShaderSemantic(AZ::Name("POSITION")),
-                    AZ::Name(),
-                    {
-                        positionBuffer,
-                        AZ::RHI::BufferViewDescriptor::CreateStructured(0, static_cast<uint32_t>(positionCount / 3), sizeof(float) * 3)
-                    }
-                );
-            }
             lodCreator.EndMesh();
+        }
+
+        // overload of Add - here a new index/position buffer is created for the new data instead of potentially reusing an existing buffer
+        void Add(
+            AZ::RPI::ModelLodAssetCreator& lodCreator,
+            const float* positions,
+            size_t positionCount,
+            size_t positionOffset,
+            const uint32_t* indices,
+            size_t indexCount,
+            size_t indexOffset)
+        {
+            AZ::Data::Asset<AZ::RPI::BufferAsset> indexBuffer = BuildTestBuffer(aznumeric_cast<uint32_t>(indexCount), sizeof(uint32_t));
+            AZ::Data::Asset<AZ::RPI::BufferAsset> positionBuffer =
+                BuildTestBuffer(aznumeric_cast<uint32_t>(positionCount / 3), sizeof(float) * 3);
 
+            Add(lodCreator, positions, positionCount, positionOffset, positionBuffer, indices, indexCount, indexOffset, indexBuffer);
+        }
+
+        // complete the asset lot creation process
```

Comment:
is "lot" correct here?

Reply:
You're right it's not, this should be `load`, will fix :+1:

Reply:
actually it should be `lod` 🤦‍♂️